OpenRA.Mods.Common/Traits/Attack/AttackBase.cs

```diff
@@ -70,7 +70,7 @@ public AttackBase(Actor self, AttackBaseInfo info)
 
 		protected virtual bool CanAttack(Actor self, Target target)
 		{
-			if (!self.IsInWorld || IsTraitDisabled)
+			if (!self.IsInWorld || self.IsDisabled() || IsTraitDisabled)
```

Comment:
Whilst you're here - moving the `!target.IsValidFor(self)` check before the `!HasAnyValidWeapons(target)` check would also be worthwhile as the first is cheaper than the second.